src/main/java/org/cryptomator/ui/traymenu/AwtTrayMenuController.java

```diff
@@ -58,6 +61,22 @@ public void updateTrayMenu(List<TrayMenuItem> items) {
 		addChildren(menu, items);
 	}
 
+
+	@Override
+	public void onBeforeOpenMenu(Runnable listener) {
+		try {
+			this.trayIcon.addMouseListener(new MouseAdapter() {
+				@Override
+				public void mouseClicked(MouseEvent e) {
+					listener.run();
+				}
+			});
+		} catch (Exception e) {
```

Comment:
Can you narrow this down to the expected exception types? I guess this is mostly for checking whether the tray icon is non-null? In this case, add an upfront check to the beginning of the method. You can use Guava's `Preconditions` utility.

Reply:
done!